### ISSUE TITLE
Shared folder types and extending interface in Vue component setup error

### ISSUE THREAD

**Issue by bnason** [nuxt/nuxt#30213]
### Environment

Nuxt project info:                                                                                                                                                     1:33:35 PM

------------------------------
- Operating System: Linux
- Node Version:     v22.3.0
- Nuxt Version:     3.14.1592
- CLI Version:      3.16.0
- Nitro Version:    2.10.4
- Package Manager:  npm@10.8.1
- Builder:          -
- User Config:      default
- Runtime Modules:  @nuxt/icon@1.9.1, @nuxt/image@1.8.1, @nuxtjs/tailwindcss@6.12.2, dayjs-nuxt@2.1.11, shadcn-nuxt@0.11.3, @nasontech/nuxt-logger@0.0.2, @nasontech/nuxt-update-available@0.0.2, @nasontech/nuxt-healthcheck@0.0.2, @nasontech/nuxt-cors@0.0.1
- Build Modules:    -
------------------------------

### Reproduction

https://codesandbox.io/p/devbox/lrzygp

### Describe the bug

Attempting to use an interface defined in the `./shared/types` folder to extend another interface results in the following error: `Internal server error: [@vue/compiler-sfc] Failed to resolve extends base type.`

Example:
```
<script setup lang="ts">
interface Props extends BudgetCategory {}

defineProps<Props>()
</script>

<template>
	<div>
		<BudgetItem v-for="item in items" v-bind="item" />
	</div>
</template>

```

### Logs

```shell-script

```

**Comment by OskarLebuda** [comment]
I don't think this is good idea from a performance reasons. The system had to discover all types from shared and add it. There can be types for both server and client. When you will try to import server types on client site - app will crash.

**Comment by OrbisK** [comment]
Reproduction ported to v4: https://stackblitz.com/edit/github-d8t47mhx

**Comment by danielroe** [comment]
resolved in latest vue version 🎉
